tests/test_analysis.py

```diff
@@ -420,50 +420,68 @@ def test_mix_scenarios():
-    updated_values = get_parameters_for_improved_healthsystem_and_healthcare_seeking(
-        resourcefilepath=resourcefilepath,
-        max_healthsystem_function=True,
-        max_healthcare_seeking=True
-    )
+    def check_parameters(sim, phase_of_simulation) -> None:
+        """Check that the paramter values in the simulation currently match expectation of that phase of the
+        simulation (0=first phase before the change, 1=second phase after the change)."""
+
+        updated_values = get_parameters_for_improved_healthsystem_and_healthcare_seeking(
+            resourcefilepath=resourcefilepath,
+            max_healthsystem_function=max_healthsystem_function[phase_of_simulation],
+            max_healthcare_seeking=max_healthcare_seeking[phase_of_simulation]
+        )
+
+        for module, param in updated_values.items():
+            for name, target_value in param.items():
 
-    for module, param in updated_values.items():
-        for name, target_value in param.items():
+                actual = sim.modules[module].parameters[name]
 
-            actual = sim.modules[module].parameters[name]
+                if isinstance(target_value, pd.Series):
+                    pd.testing.assert_series_equal(target_value, actual)
+                elif isinstance(target_value, pd.DataFrame):
+                    pd.testing.assert_frame_equal(target_value, actual)
+                elif isinstance(target_value, list):
+                    assert all([t == v for t, v in zip(target_value, actual)])
+                else:
+                    assert target_value == actual
 
-            if isinstance(target_value, pd.Series):
-                pd.testing.assert_series_equal(target_value, actual)
-            elif isinstance(target_value, pd.DataFrame):
-                pd.testing.assert_frame_equal(target_value, actual)
-            elif isinstance(target_value, list):
-                assert all([t == v for t, v in zip(target_value, actual)])
-            else:
-                assert target_value == actual
+        # Spot check for health care seeking being forced to occur for all symptoms
+        hcs = sim.modules["HealthSeekingBehaviour"].force_any_symptom_to_lead_to_healthcareseeking
+        assert isinstance(hcs, bool) and (hcs is max_healthcare_seeking[phase_of_simulation])
 
-    # Spot check for health care seeking being forced to occur for all symptoms
-    hcs = sim.modules['HealthSeekingBehaviour'].force_any_symptom_to_lead_to_healthcareseeking
-    assert isinstance(hcs, bool) and hcs
+    # Check that all the parameter values in the simulation are updated to be the value expected at the beginning of
+    # the simulation.
+    check_parameters(sim, 0)
 
+    # Run the simulation until after the date of the change and then check the values have been updated.
+    sim.simulate(end_date=Date(year_of_change, 2, 1))
+    check_parameters(sim, 1)
```

Comment:
As I have said above I would avoid manually setting `phase_of_simulation` argument. Would be great if we used `phase_of_simulation`  variable defined in scenario switcher